File: GC.Domain/Gardens/Sectors/GardenSector.cs
using System;

namespace GC.Domain.Gardens.Sectors
{
    public class GardenSector
    {
        public Guid Id { get; }
        public Int32 SectorNumber { get; }
        public String ElectricityNumber { get; }
        public Double NumberOfAcres { get; }
        public String CadastralNumber { get; }

        public GardenSector(Guid id, Int32 sectorNumber, String electricityNumber, Double numberOfAcres, String cadastralNumber)
        {
            Id = id;
            SectorNumber = sectorNumber;
            ElectricityNumber = electricityNumber;
            NumberOfAcres = numberOfAcres;
            CadastralNumber = cadastralNumber;
        }
    }
}

File: GC.WebSpace/ClientApp/domain/gardens/gardenSectors/gardenSector.ts

export class GardenSector {
    constructor(
        public readonly id: string,
        public readonly sectorNumber: number,
        public readonly electricityNumber: string,
        public readonly numberOfAcres: number,
        public readonly cadastralNumber: string
    ) { }
}

export namespace GardenSector {
    export function toModel(value: any) {
        return new GardenSector(value.id, value.sectorNumber, value.electricityNumber, value.numberOfAcres, value.cadastralNumber);
    }

    export function toModels(values: any[]) {
        return values.map(toModel);
    }
}